### ISSUE TITLE
Create a `4.0` branch

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/extension-cookiecutter-ts#231]
### Problem

It would be great to start encouraging extension authors to create / test their extensions with the JupyterLab 4 and Notebook 7 pre-releases using this cookiecutter.

### Proposed Solution

- Create a new `4.0` branch to update the packages to the latest 4.0 pre-release.
- Document it in the README: `cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts --checkout v4.0`.

### Additional context

The `3.0` branch would still remain the default branch for now. Changes landing on `3.0` could then be backported to `4.0`.

**Comment by jtpio** [comment]
The `4.0` branch has been created: https://github.com/jupyterlab/extension-cookiecutter-ts/tree/4.0

**Comment by jtpio** [comment]
We can backport changes landing on the `3.0` to the `4.0` if any, until `4.0` becomes the default branch.

**Comment by ragokan** [comment]
I also see that it is created but it actually does not work, weirdly

```bash
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts --checkout v4.0
You've downloaded /Users/ragokan/.cookiecutters/extension-cookiecutter-ts before. Is it okay to delete and re-download it? [yes]: yes
The v4.0 branch of repository https://github.com/jupyterlab/extension-cookiecutter-ts could not found, have you made a typo?
```

**Comment by jtpio** [comment]
@ragokan you would have to checkout `4.0`:

```
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts --checkout 4.0
```

**Comment by jtpio** [comment]
Closing as fixed, the `4.0` is now updated to use the latest packages: https://github.com/jupyterlab/extension-cookiecutter-ts/pull/234
